### PR TITLE
Simulated Backend: Modify simulated backend with GasPrice

### PR DESCRIPTION
**Pull request by KimKyungup**
## Proposed changes
This PR add a new simulated backend with gasPrice for testing with non-zero unit price.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [x] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: accounts/abi/bind/backends/simulated.go [hunk 1]
```diff
@@ -68,8 +68,22 @@ type SimulatedBackend struct {
 // NewSimulatedBackend creates a new binding backend using a simulated blockchain
 // for testing purposes.
 func NewSimulatedBackend(alloc blockchain.GenesisAlloc) *SimulatedBackend {
-	database := database.NewMemoryDBManager()
 	genesis := blockchain.Genesis{Config: params.AllGxhashProtocolChanges, Alloc: alloc}
+	return newSimulatedBackend(genesis)
+}
+
+// NewSimulatedBackend creates a new binding backend using a simulated blockchain with given unitPrice.
```
Comment (jimni1222) [inline accounts/abi/bind/backends/simulated.go]:
```suggestion
// NewSimulatedBackendWithGasPrice creates a new binding backend using a simulated blockchain with given unitPrice.
```

File: accounts/abi/bind/backends/simulated.go [hunk 2]
```diff
@@ -68,8 +68,22 @@ type SimulatedBackend struct {
...
+// for testing purposes.
+func NewSimulatedBackendWithGasPrice(alloc blockchain.GenesisAlloc, unitPrice uint64) *SimulatedBackend {
+	genesis := blockchain.Genesis{Config: params.AllGxhashProtocolChanges, Alloc: alloc}
+	genesis.Config.UnitPrice = unitPrice
+
+	return newSimulatedBackend(genesis)
+}
+
+// newSimulatedBackend creates a new binding backend with the genesis.
```
Comment (aidan-kwon) [inline accounts/abi/bind/backends/simulated.go]:
```suggestion
// newSimulatedBackend creates a new binding backend with a genesis.
```
